platform/kube/conversion.go

```diff
@@ -82,11 +82,16 @@ func convertPort(port v1.ServicePort) *model.Port {
 
 func convertService(svc v1.Service) *model.Service {
 	ports := make([]*model.Port, 0)
-	addr := ""
+	addr, external := "", false
 	if svc.Spec.ClusterIP != "" && svc.Spec.ClusterIP != v1.ClusterIPNone {
 		addr = svc.Spec.ClusterIP
 	}
 
+	if svc.Spec.Type == v1.ServiceTypeExternalName && svc.Spec.ExternalName != "" {
+		addr = svc.Spec.ExternalName
```

Comment:
Do you mean adding it to model.Service ? If so, then it simply conflates the model. Services already have a hostname in the model and there is nothing in the model that says a service is in cluster or not. So adding an ExternalName makes little sense from a generic model perspective. 

If anything, I would add a type field to the model.Service, that can qualify a service as either "internal/external", deployment mode as a boolean "user-facing or not". This would let ingress/gateway Envoy pick up only user-facing services of internal Type, non-k8s environments), egress Envoy pick external services while other envoys can pickup internal services and external service (with a host rewrite to egress envoy).